staff/fede-gomez/js-ayay/ayay.js

```diff
@@ -11,26 +11,38 @@ function Ayay() {
 
 Ayay.prototype.push = function(element) {
     this[this.length] = element;
+    if(element) this.length++;
+    return this.length;
+};
+
+Ayay.prototype.forEach = function(callback) {
+
+    if (typeof callback !== 'function') throw new TypeError(callback + ' is not a function');
 
-    this.length++;
+    for (var i = 0; i < this.length; i++) callback(this[i], i, this);
 };
 
 Ayay.prototype.pop = function() {
     
-    poppedElem = this[this.length];
+    poppedElem = this[this.length-1];
     this.length--;
-    this[this.length] = undefined;
+    delete this[this.length];
     return poppedElem;
 };
 
-Ayay.prototype.forEach = function(callback) {
-    if (typeof callback !== 'function') throw new TypeError(callback + ' is not a function');
+Ayay.prototype.map = function(callback) {
+    var newAyay = new Ayay;
 
-    for (var i = 0; i < this.length; i++) callback(this[i], i, this);
-};
+    //We have to iterate through the original ayay and return the values that passed the criteria in the callback
+    
+    for(var i=0; i<this.length; i++){
+        if(callback(this[i], i)){newAyay.push = this[i]}
```

Comment:
maybe it's easier this way: 
for(var i = 0; i<this.length; i++){
        newAy.push(callback(this[i], i));
    }